Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -1594,28 +1600,37 @@ bool CStaticFunctionDefinitions::SetElementModel(CElement* pElement, unsigned sh
                 return false;
             if (!CVehicleManager::IsValidModel(usModel))
                 return false;
+            unsigned short usOldModel = pVehicle->GetModel();      // Get the old model
             CLuaArguments Arguments;
-            Arguments.PushNumber(pVehicle->GetModel());            // Get the old model
+            Arguments.PushNumber(usOldModel);
             pVehicle->SetModel(usModel);                           // Set the new model
             Arguments.PushNumber(usModel);                         // Get the new model
-            pVehicle->CallEvent("onElementModelChange", Arguments);
-
-            // Check for any passengers above the max seat list
-            unsigned char ucMaxPassengers = pVehicle->GetMaxPassengers();
-            CLogger::DebugPrintf("Max passengers = %u\n", ucMaxPassengers);
-            unsigned char i = 0;
-            for (; i < MAX_VEHICLE_SEATS; i++)
-            {
-                // Got a player in this seat and is it bigger than the supported amount
-                // of seats in this new vehicle
-                CPed* pPed = pVehicle->GetOccupant(i);
-                if (pPed && IS_PLAYER(pPed) && (i > ucMaxPassengers))
+            if (!pVehicle->CallEvent("onElementModelChange", Arguments))
+            {
+                // Change canceled
+                pVehicle->SetModel(usOldModel);
+                return false;

```

Comment:
the `else` below is not required because of short-circuiting here

Reply:
Yes, this is unnecessary